lib/sqs-handler.js

```diff
@@ -50,21 +59,69 @@ module.exports = class SQSHandler {
 
 		SQSEventStruct(event);
 
-		const consumer = new Consumer();
-		if(consumer.handlesBatch(event)) {
-			await consumer.processBatch(event.Records.map(record => {
-				const parsedRecord = this.parseRecord(record);
-				const logger = new LogTransport(record.messageId);
-				parsedRecord[Symbol.for('logger')] = logger;
-				return parsedRecord;
-			}));
-		} else {
-			await Promise.all(event.Records.map(record => {
-				const logger = new LogTransport(record.messageId);
-				return consumer.processSingleRecord(this.parseRecord(record), logger);
-			}));
+		const isBatch = (new Consumer()).handlesBatch(event);
+		if(isBatch)
+			await this.handleBatch(Consumer, event);
+		else
+			await this.handleSingle(Consumer, event);
+	}
+
+	/**
+	 * Process records in batch but splitting by client or without client
+	 *
+	 * @param {import('./sqs-consumer')} Consumer
+	 * @param {SQSEvent} event
+	 * @param {boolean} isBatch
+	 * @returns {Promise}
```

Comment:
Solo para dejar en claro que el valor al que resuelve va a ser ignorado...

```suggestion
	 * @returns {Promise<void>}
```